### DISCUSSION TITLE
Getting weird websocket errors

### DISCUSSION THREAD

**Discussion by rogrile** [nuxt/nuxt#27779] [General]
Hey, I've just gotten back to working on my side project. Tried to update existing repo but I kept seeing errors and devtools wasn't working.

I have started a new repo from scratch. Updated npm and node and I am using latest version of Nuxt 3.

I keep getting this error: `WebSocket connection to 'ws://localhost:24678/_nuxt/' failed`

This is working on Stackblitz. I am on MacOS Sonoma 14.5.

So fresh install just keeps throwing this error in the console:
```
client.ts:339 
 GET http://localhost:24678/_nuxt/ net::ERR_CONNECTION_REFUSED
client.ts:339 
 GET http://localhost:24678/_nuxt/ net::ERR_CONNECTION_REFUSED
client.ts:339 
 GET http://localhost:24678/_nuxt/ net::ERR_CONNECTION_REFUSED
client.ts:339 
 GET http://localhost:24678/_nuxt/ net::ERR_CONNECTION_REFUSED
```

I cannot find any info about this anywhere and I have no idea how to solve it. There is nothing else running on that port locally. I am just starting the new project using `npx nuxi@latest init nuxt-test ` and running it with `npm run dev`. I am doing nothing else and I cannot change the behaviour  regardless of what I've tried (such as disabling HMR from vite)

**Comment by rogrile** [comment]
Looks like the only way to solve it is to do this:
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  devtools: { enabled: true },
  vite: {
    server: {
      hmr: {
        protocol: "http",
        host: 'localhost',
        clientPort: 3000,
        port: 3000,
      },
    },
  },
});


Not sure how this is going to affect it once I try to deploy it in my prod environment

**Reply by SergkeiM** [reply]
Have exactly same issue, by specifing this ports ( which are defaults) works fine.

Actually by only specifing `clientPort: 3000,` otherwise it tries to connect to WS on port `24678`

**Reply by yesworld** [reply]
Same issue in Chrome 
<img width="754" alt="image" src="https://github.com/nuxt/nuxt/assets/6162264/0d6fc019-97a0-4e0a-8ae3-ba949b15e993">

**Reply by hermanndettmann** [reply]
Thanks @rogrile! Setting this in `nuxt.config.ts` removed the console errors:

```
vite: {
    server: {
        hmr: {
            clientPort: 3000
        }
    }
},
```

But that feels more like a temporary fix.

**Reply by sincely** [reply]
@hermanndettmann @SergkeiM  thanks！！

**Reply by MrSunshyne** [reply]
@hermanndettmann i had the same issue after updating to latest nuxt today and this solved it for me. Thanks !

**Reply by hermanndettmann** [reply]
Good to hear! You‘re welcome!

**Reply by JhumanJ** [reply]
Same! Setting client port to 3000 fixed it

**Reply by vishakha99-gif** [reply]
Thanks @hermanndettmann works for me..

**Reply by toto6038** [reply]
I've run into this problem after upgrading Nuxt version in my project, and the solution was simply removing node_modules and then installing dependencies again. It may not appeared to be a clever solution, but it worked anyway.
